moveit_core/robot_model/src/robot_model.cpp

```diff
@@ -293,19 +293,28 @@ void RobotModel::buildJointInfo()
     }
   }
 
+  bool link_considered[link_model_vector_.size()] = { false };
   for (std::size_t i = 0; i < link_model_vector_.size(); ++i)
   {
+    if (link_considered[i])
+      continue;
+
     LinkTransformMap associated_transforms;
     computeFixedTransforms(link_model_vector_[i], link_model_vector_[i]->getJointOriginTransform().inverse(),
                            associated_transforms);
-    if (associated_transforms.size() > 1)
+    for (auto& tf_base : associated_transforms)
     {
-      for (LinkTransformMap::iterator it = associated_transforms.begin(); it != associated_transforms.end(); ++it)
-        if (it->first != link_model_vector_[i])
+      link_considered[tf_base.first->getLinkIndex()] = true;
+      for (auto& tf_target : associated_transforms)
+      {
+        if (&tf_base != &tf_target)
         {
-          getLinkModel(it->first->getName())->addAssociatedFixedTransform(link_model_vector_[i], it->second.inverse());
-          link_model_vector_[i]->addAssociatedFixedTransform(it->first, it->second);
+          // this const-cast is equivalent to
+          // associated_transforms[tf_base.first->getLinkIndex()]
```

Comment:
This doesn't make sense at all, does it?
`associated_transforms` is probably way smaller than `tf_base.first->getLinkIndex()`.
You just want to regain write access to the "base" link, don't you?
I will add a commit to fix this comment.